state/enableha.go

```diff
@@ -34,51 +34,48 @@ func isController(mdoc *machineDoc) bool {
 
 var errControllerNotAllowed = errors.New("controller jobs specified but not allowed")
 
-func (st *State) getVotingMachineCount(info *ControllerInfo) (int, error) {
+func (st *State) getVotingControllerCount() (int, error) {
 	controllerNodesColl, closer := st.db().GetCollection(controllerNodesC)
 	defer closer()
 
-	return controllerNodesColl.Find(
-		bson.M{
-			"_id":        bson.M{"$in": info.MachineIds},
-			"wants-vote": true,
-		},
-	).Count()
+	return controllerNodesColl.Find(bson.M{"wants-vote": true}).Count()
 }
 
 // maintainControllersOps returns a set of operations that will maintain
 // the controller information when the given machine documents
 // are added to the machines collection. If currentInfo is nil,
 // there can be only one machine document and it must have
 // id 0 (this is a special case to allow adding the bootstrap machine)
-func (st *State) maintainControllersOps(newIds []string, currentInfo *ControllerInfo) ([]txn.Op, error) {
+func (st *State) maintainControllersOps(newIds []string, bootstrapOnly bool) ([]txn.Op, error) {
 	if len(newIds) == 0 {
 		return nil, nil
 	}
-	if currentInfo == nil {
+	currentControllerIds, err := st.ControllerIds()
+	if !bootstrapOnly && errors.IsNotFound(err) {
+		return nil, errors.Trace(err)
+	}
+	if err != nil {
```

Comment:
Won't this raise an error if `bootstrapOnly` is true and `errors.IsNotFound(err)`? It seems like it should continue in that case?

Reply:
turns out i didn't need the error check as the controller doc is always there